sdk/core/azure-core/src/http/log_policy.cpp

```diff
@@ -37,13 +37,32 @@ inline void AppendHeaders(
   }
 }
 
+inline std::string GetUrlWithoutQuery(Url url)
```

Comment:
This is going to be invoked when logging during each request. Let's use `std::ostringstream` - either take as a parameter (non-const ref), or build in a string stream, and then `return sstream.c_str();`.